### PR TITLE
Add pagination of results

### PR DESCRIPTION
**Pull request by MiroslavGatsanoga**

### CODE REVIEW

File: content/handlers.go [hunk 1]
```diff
@@ -64,6 +68,27 @@ func (ch *ContentByConceptHandler) GetContentByConcept(w http.ResponseWriter, r
...
+			msg := fmt.Sprintf("provided value for page, %s, could not be parsed.", pageParam)
+			logger.WithError(err).WithTransactionID(transID).WithUUID(conceptUUID).Error(msg)
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(`{"message": ` + msg + `}`))
+			return
+		}
+
+		if page < defaultPage {
+			msg := fmt.Sprintf("provided value for page should be greater than: %v", defaultPage)
+			logger.WithTransactionID(transID).WithUUID(conceptUUID).Debugf(msg)
```
Comment (mchompalova) [inline content/handlers.go]: Shouldn't we have WithError() here?

File: content/handlers_test.go [hunk 2]
```diff
@@ -119,22 +129,20 @@ func TestContentByConceptHandler_GetContentByConcept(t *testing.T) {
...
-			backendError:       nil,
 		},
 		{
 			testName:           "Backend Error returns 503",
 			conceptID:          testConceptID,
 			contentList:        []string{testContentUUID},
 			expectedStatusCode: 503,
 			expectedBody:       "{\"message\": Backend error returning content for concept with uuid 44129750-7616-11e8-b45a-da24cd01f044}",
-			backendError:       errors.New("There was a problem"),
+			backendError:       errors.New("there was a problem"),
```
Comment (mchompalova) [inline content/handlers_test.go]: Messages start with a capital letter elsewhere

File: content/service.go [hunk 3]
```diff
@@ -51,8 +52,11 @@ func (cd ConceptService) GetContentForConcept(conceptUUID string, params Request
 		whereClause = " WHERE c.publishedDateEpoch > {fromDate} AND c.publishedDateEpoch < {toDate}"
 	}
 
+	skipCount := (params.page - 1) * params.contentLimit
```
Comment (mchompalova) [inline content/service.go]: It's not quite clear what the skipCount parameter is used for; can we add a comment here?

File: content/service_test.go [hunk 4]
```diff
@@ -230,14 +232,61 @@ func TestContentIsReturnedFromAllLeafNodesOfConcordanceWithDateRestrictions(t *t
...
+			}
+
+			assert.NoError(err, "Unexpected error for concept %s", uuid)
+			assert.Equal(pageSize, len(pageContents), "Didn't get the right number of page items, content=%s", pageContents)
+
+			page++
+			allContent = append(allContent, pageContents...)
+		}
+
+		assert.Equal(4, len(allContent), "Didn't get the right number of content items, content=%s", allContent)
```
Comment (mchompalova) [inline content/service_test.go]: I don't understand why we need the "Didn't get" message here.

File: content/handlers_test.go [hunk 5]
```diff
@@ -51,48 +63,48 @@ func TestContentByConceptHandler_GetContentByConcept(t *testing.T) {
...
 			expectedBody:       `{"message": "ID extracted from request URL was not valid uuid"}`,
-			backendError:       nil,
+		},
+		{
+			testName:           "Bad Request: query param 'page' is invalid",
+			conceptID:          testConceptID,
+			contentList:        []string{testContentUUID},
+			page:               "null",
+			expectedStatusCode: 400,
+			expectedBody:       "{\"message\": provided value for page, null, could not be parsed.}",
```
Comment (mchompalova) [inline content/handlers_test.go]: Can we add a test for the case when page < defaultPage?

File: content/handlers.go [hunk 6]
```diff
@@ -64,6 +68,27 @@ func (ch *ContentByConceptHandler) GetContentByConcept(w http.ResponseWriter, r
 		return
 	}
 
+	page := defaultPage
+	pageParam := m.Get("page")
+	if pageParam != "" {
+		page, err = strconv.Atoi(pageParam)
+		if err != nil {
+			msg := fmt.Sprintf("provided value for page, %s, could not be parsed.", pageParam)
```
Comment (mchompalova) [inline content/handlers.go]: Can you capitalize the error message?

File: content/handlers_test.go (hunk 2 above)
Comment (MiroslavGatsanoga) [reply content/handlers_test.go]: It actually is considered good practice for error strings to not be capitalized: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

File: content/service.go (hunk 3 above)
Comment (MiroslavGatsanoga) [reply content/service.go]: Ok, I'll add a comment.

File: content/handlers_test.go (hunk 5 above)
Comment (MiroslavGatsanoga) [reply content/handlers_test.go]: Sure, I have missed that case. Nice catch!

File: content/handlers.go (hunk 6 above)
Comment (MiroslavGatsanoga) [reply content/handlers.go]: As mentioned above, error strings should not be capitalized.

File: content/service_test.go (hunk 4 above)
Comment (MiroslavGatsanoga) [reply content/service_test.go]: Do you think we can use a better wording for the message here? I am open to suggestions.

Comment (mchompalova) [reply content/service_test.go]: It seems that we need this message for debugging purposes.

**Pull request merged**